### PR TITLE
Ocp2.6 Hardware Component Log 

### PR DESCRIPTION
**Pull request by sc108-lee**
Based #2570 spec modification based ocp2.6

Log Identifier C6h
Hardware Component Log Size (bytes)
Component Identifier added
0x000C Born on Date. The date on which the device was manufactured.
ASCII string format is MMDDYYYY.

### CODE REVIEW

File: plugins/ocp/ocp-hardware-component-log.c [hunk 1]
```diff
@@ -154,6 +154,8 @@ const char *hwcomp_id_to_string(__u32 id)
 		return "Country of Origin";
 	case HWCOMP_ID_HW_REV:
 		return "Global Device Hardware Revision";
+	case HWCOMP_ID_BORN_ON_DATE:
+		return "Born on Date";
```
Comment (ikegami-t) [inline plugins/ocp/ocp-hardware-component-log.c:158]: The command option comp-id is not added.
```c
	OPT_VALS(id) = {
		VAL_LONG("asic", HWCOMP_ID_ASIC),
		...
		VAL_LONG("hw-rev", HWCOMP_ID_HW_REV),
		VAL_LONG("vendor", HWCOMP_ID_VENDOR),
		VAL_END()
	};
```

**Review by ikegami-t — COMMENTED** [review]
By the way only changed the hardware component log size and added the component ID born on date for the log page by OCP verion 2.6? Is there any documentation or information for the changes version 2.6? Should we compare the documentation with the older version documentation? (I did this before to check the OCP 2.5 changes..)

**Comment by sc108-lee** [comment]
https://www.opencompute.org/events/past-events/2024-ocp-storage-tech-talk

I could find a documentation, but I don't think this doc checks everything. (since there is no Hardware Component Log change)
check it out.

**Comment by ikegami-t** [comment]
The changes look good. Thank you so much for your information. Just watched the OCP video for several 10 minutes and very interesting so will check more later since the play time is long.

**Comment by igaw** [comment]
@sc108-lee could you rebase this PR? Looks like it conflicts with the merge from #2570. Thanks

**Comment by sc108-lee** [comment]
@igaw  rebased :)

**Pull request merged**